aquarius/ddo_checker/shacl_schemas/v4/remote_4.0.0.ttl

```diff
@@ -0,0 +1,351 @@
+    ] ;
+    sh:property [
+        sh:path schema:serviceEndpoint ;
+        sh:datatype xsd:string ;
+        sh:pattern "^(.*)$" ;
+        sh:minCount 1;
+        sh:maxCount 1;
+    ] ;
+    sh:property [
+        sh:path schema:files;
+        sh:datatype xsd:string ;
+        sh:pattern "^(.*)$" ;
+        sh:minCount 1;
+        sh:maxCount 1;
+    ] ;
+    sh:property [
+        sh:path schema:timeout ;
+        sh:datatype xsd:integer ;
+        sh:pattern "^(.*)$" ;
+        sh:minCount 1;
+        sh:maxCount 1;
+    ] ;
+    sh:property [
+        sh:path schema:compute;
+        sh:node schema:ComputeShape ;
+        sh:maxCount 1;
+    ] ; .
+
+
+schema:ComputeShape
+    sh:targetClass schema:Compute ;
+    sh:property [
+        sh:path schema:namespace ;
+        sh:datatype xsd:string ;
+        sh:pattern "^(.*)$" ;
+        sh:minCount 1;
+        sh:maxCount 1;
+    ] ;
+    sh:property [
+        sh:path schema:cpus ;
+        sh:datatype xsd:integer ;
+        sh:maxCount 1;
+    ] ;
+    sh:property [
+        sh:path schema:gpus ;
+        sh:datatype xsd:integer ;
+        sh:maxCount 1;
+    ] ;
+    sh:property [
+        sh:path schema:gpuType ;
+        sh:datatype xsd:string ;
+        sh:pattern "^(.*)$" ;
+        sh:maxCount 1;
+    ] ;
+    sh:property [
+        sh:path schema:memory ;
+        sh:datatype xsd:string ;
+        sh:pattern "^(.*)$" ;
+        sh:maxCount 1;
+    ] ;
+    sh:property [
+        sh:path schema:volumeSize;
+        sh:datatype xsd:string ;
+        sh:pattern "^(.*)$" ;
+        sh:maxCount 1;
+    ] ;
+    sh:property [
+        sh:path schema:allowRawAlgorithm ;
+        sh:datatype xsd:boolean ;
+        sh:minCount 1;
+        sh:maxCount 1;
+    ] ;
+    sh:property [
+        sh:path schema:allowNetworkAccess ;
+        sh:datatype xsd:boolean ;
+        sh:minCount 1;
+        sh:maxCount 1;
+    ] ;
+    sh:property [
+        sh:path schema:publisherTrustedAlgorithmPublishers ;
+        sh:datatype xsd:string ;
+        sh:pattern "^(.*)$" ;
+    ] ;
+    sh:property [
+        sh:path schema:publisherTrustedAlgorithms;
+        sh:node schema:TrustedAlgoShape ;
+    ] ; .
+
+
+schema:TrustedAlgoShape
+    sh:targetClass schema:TrustedAlgo;
+    sh:property [
+        sh:path schema:did ;
+        sh:datatype xsd:string ;
+        sh:pattern "^(.*)$" ;
+        sh:minCount 1;
+        sh:maxCount 1;
+    ] ;
+    sh:property [
+        sh:path schema:filesChecksum ;
+        sh:datatype xsd:string ;
+        sh:pattern "^(.*)$" ;
+        sh:minCount 1;
+        sh:maxCount 1;
+    ] ;
+    sh:property [
+        sh:path schema:containerSectionChecksum ;
+        sh:datatype xsd:string ;
+        sh:pattern "^(.*)$" ;
+        sh:minCount 1;
+        sh:maxCount 1;
+    ] ; .
+
+
+schema:CredentialsShape
+    sh:targetClass schema:Credentials;
+    sh:property [
+        sh:path schema:deny ;
+        sh:node schema:CredentialsItemShape;
+        sh:maxCount 1;
```

Comment:
Is this right?  I thought there could be multiple allow credentials and multiple deny credentials?

Reply:
just one "allow" key, with multiple items (see credentials item shape), just one "deny" key